### ISSUE TITLE
Twisted version string breaks backward compatibility in 16.5.0rc1

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8863]
|[<img alt="hellais's avatar" src="https://avatars.githubusercontent.com/u/424620?s=50" width="50" height="50">](https://github.com/hellais)| @hellais reported|
|-|-|
|Trac ID|trac#8863|
|Type|release blocker: regression|
|Created|2016-10-16 20:13:38Z|
|Branch|https://github.com/twisted/twisted/tree/8863-version-caps|

Since https://github.com/twisted/twisted/commit/4bb0e5a93b57db06202e03388695ee6995bb4459 the object for twisted (`twisted.version`) has changed from being "twisted" (all lowercase) to "Twisted" first letter uppercase.

As a result of this many pieces of software that need to determine the version of twisted and to do so use `twisted.version` since 16.4.1 will fail with the following exception:

```
File "/usr/local/lib/python2.7/dist-packages/incremental-16.10.0-py2.7.egg/incremental/__init__.py", line 341, in __cmp__
% (self.package, other.package))
incremental.IncomparableVersions: 'Twisted' != 'twisted'
```

An example is the socks library txsocksx. Look at here for the relevant pieces of code:

https://github.com/habnabit/txsocksx/blob/master/txsocksx/http.py#L24

Minimal reproducibility of the bug:

```
pip install twisted==16.4.1
python -c 'import twisted;from twisted.python.versions import Version;print twisted.version > Version("twisted", 1, 1, 0)'
# prints true

pip install twisted==16.5.0rc1
python -c 'import twisted;from twisted.python.versions import Version;print twisted.version > Version("twisted", 1, 1, 0)'
# Triggers exception
```

I think the fix is just a matter of replacing the name of the package with "twisted" (all lowercase).

<details><summary>Searchable metadata</summary>

```
trac-id__8863 8863
type__release_blocker__regression release blocker: regression
reporter__hellais hellais
priority__normal normal
milestone__None None
branch__8863_version_caps 8863-version-caps
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1476648818633991 1476648818633991
changetime__1477196042603604 1477196042603604
version__None None
owner__Amber_Brown__hawkowl_____ Amber Brown <hawkowl@...>
cc__habnabit
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

This one should definitely be a release blocker.  Thank you, hellais, for actually testing within the release window!!! :-D

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Okay, should be good (except for GitHub making the appveyor builds not run).

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_5"></a>|@markrwilliams commented|
|-|-|

https://github.com/twisted/twisted/pull/567#pullrequestreview-5343974

Merge when green!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set owner to @hawkowl<br>@hawkowl set status to `closed`|
|-|-|

In changeset dc6be3bebdaf99e6f9db5e5b6119b0a620624078
```
#!CommitTicketReference repository="" revision="dc6be3bebdaf99e6f9db5e5b6119b0a620624078"
Merge 8863-version-caps: Make t.p.versions.Version case insensitive

Author: hawkowl
Reviewer: markrwilliams
Fixes: #8863
```
